engine/src/main/scala/cromwell/webservice/routes/CromwellApiService.scala

```diff
@@ -186,7 +185,23 @@ trait CromwellApiService extends HttpInstrumentation with MetadataRouteSupport w
     WorkflowSubmitResponse(workflowId.toString, workflowState.toString)
   }
 
-  private def submitRequest(formData: Multipart.FormData, isSingleSubmission: Boolean): Route = {
+  def standardSuccessHandler: PartialFunction[WorkflowStoreSubmitActorResponse, Route] = {
+    case WorkflowStoreSubmitActor.WorkflowSubmittedToStore(workflowId, _) => completeResponse(StatusCodes.Created, WorkflowSubmitResponse(workflowId.toString, WorkflowSubmitted.toString), Seq.empty[String])
+    case WorkflowStoreSubmitActor.WorkflowsBatchSubmittedToStore(workflowIds, _) => completeResponse(StatusCodes.Created, workflowIds.toList.map(x => WorkflowSubmitResponse(x.toString, WorkflowSubmitted.toString)),  Seq.empty[String])
+    case WorkflowStoreSubmitActor.WorkflowSubmitFailed(throwable) => throwable.failRequest(StatusCodes.BadRequest)
+  }
+
+  def standardErrorHandler: PartialFunction[Throwable, Route] = {
+    case _: AskTimeoutException if CromwellShutdown.shutdownInProgress() => serviceShuttingDownResponse
+    case e: TimeoutException => e.failRequest(StatusCodes.ServiceUnavailable)
+    case e: Exception => e.errorRequest(StatusCodes.InternalServerError)
+  }
+
+  def submitRequest(formData: Multipart.FormData,
+                    isSingleSubmission: Boolean,
+                    successHandler: PartialFunction[WorkflowStoreSubmitActorResponse, Route] = standardSuccessHandler,
+                    errorHandler: PartialFunction[Throwable, Route] = standardErrorHandler
+                   ): Route = {
```

Comment:
It's not clear to me why this function is changing. Shouldn't low-level workflow submission be identical whether the original source came in through WES or the Cromwell API?

Reply:
I had this question myself. Without these handlers, a workflow is submitted just fine with now obvious errors. The reasoning for this was to create a "generalized" way of handling success/failures, and then for Wes, "translating" these handlers into something Wes understands, but I'm not sure it's necessary.